torch/_dynamo/variables/functions.py

```diff
@@ -328,6 +330,63 @@ def call_function(
         return super().call_function(tx, args, kwargs)
 
 
+class FunctionDecoratedByContextlibContextManagerVariable(BaseUserFunctionVariable):
+    """functions that behaves like iterators
+
+    .. note::
+
+        This is only used when the function is annotated with @contextlib.contextmanager
+    """
+
```

Comment:
@guilhermeleobas didn't we have some code before to skip the `__enter__` frame if a graph break happened? What happened to it?

Reply:
Here: "We don't actually need to skip any code. There's a check above which raises unsupported if code is a generator. I had previously removed this check, but re-introduced it a few commits ago."

https://github.com/pytorch/pytorch/pull/136033#discussion_r1841119033